### PR TITLE
add suspects to block

### PR DESCRIPTION
**Pull request by HagarMeir**
Change-Id: I36b2ebbbf8881b024843741e1b5df260f14abb82
Signed-off-by: Hagar Meir <hagar.meir@ibm.com>

### CODE REVIEW

File: orderer/consensus/smartbft/util.go [hunk 1]
```diff
@@ -719,3 +719,39 @@ type asn1Header struct {
...
+	}
+	return noDupsList
+}
+
+// agreedSuspects gets a list of all suspects (a concatenation of all suspects lists)
+// it returns the suspects that appear at least f+1 times in the given least
+func agreedSuspects(allSuspects []int32, f int32) []int32 {
+	votes := make(map[int32]int32)
+
+	for _, s := range allSuspects {
```
Comment (yacovm) [inline orderer/consensus/smartbft/util.go:741]: I think that just doing `votes[s]++` should work, have you tried it?

File: orderer/consensus/smartbft/util.go [hunk 2]
```diff
@@ -719,3 +719,39 @@ type asn1Header struct {
...
+		if _, ok := votes[s]; !ok {
+			votes[s] = 1
+			continue
+		}
+		votes[s]++
+	}
+
+	var agreedSuspects []int32
+	for s, v := range votes {
```
Comment (yacovm) [inline orderer/consensus/smartbft/util.go:746]: Is it possible to returned agreedSuspects in a sorted manner?

**Review by yacovm — CHANGES_REQUESTED** [review]
Hi. 

But what about the verification? We also need to verify that the pre-prepare contains a valid list.

**Comment by HagarMeir** [comment]
> Hi.
> 
> But what about the verification? We also need to verify that the pre-prepare contains a valid list.

of course, will be in the next PR

**Review by yacovm — DISMISSED** [review]

**Comment by yacovm** [comment]
> > Hi.
> > But what about the verification? We also need to verify that the pre-prepare contains a valid list.
> 
> of course, will be in the next PR

`2021-04-20T09:56:09.1088217Z [e][OrdererOrg.orderer2] 2021-04-20 09:54:57.666 UTC [orderer.consensus.smartbft.consensus] verifyVote -> WARN 73e Couldn't verify 4's signature: signature's OrdererBlockMetadata and OrdererBlockMetadata extracted from block do not match channel=systemchannel`

The code checks that the metadata is updated in a legal way, and you need to update its expectations.

**Pull request merged**